app/data/ContributionData.scala

```diff
@@ -0,0 +1,114 @@
+package data
+
+import anorm._
+import data.AnormMappings._
+import models.{ContributionMetaData, Contributor, PaymentHook}
+import play.api.db.Database
+
+class ContributionData(db: Database) {
+  def insertPaymentHook(paymentHook: PaymentHook): Unit = {
+    db.withConnection(autocommit = true) { implicit conn =>
+      val request = SQL"""
+        INSERT INTO payment_hooks(
+          contributionid,
+          paymentid,
+          provider,
+          created,
+          currency,
+          cardcountry,
+          amount,
+          convertedamount,
+          status,
+          email
+        ) VALUES (
+          ${paymentHook.contributionId}::uuid,
+          ${paymentHook.paymentId},
+          ${paymentHook.provider}::paymentProvider,
+          ${paymentHook.created},
+          ${paymentHook.currency},
+          ${paymentHook.cardCountry},
+          ${paymentHook.amount},
+          ${paymentHook.convertedAmount},
+          ${paymentHook.status}::paymentStatus,
+          ${paymentHook.email}
+        ) ON CONFLICT(contributionId) DO
+        UPDATE SET
+          contributionid = excluded.contributionid,
+          paymentid = excluded.paymentid,
+          provider = excluded.provider,
+          created = excluded.created,
+          currency = excluded.currency,
+          cardcountry = excluded.cardcountry,
+          amount = excluded.amount,
+          convertedamount = excluded.convertedamount,
+          status = excluded.status,
+          email = excluded.email"""
+      request.execute()
+    }
+  }
+
+  def insertPaymentMetaData(pmd: ContributionMetaData): Unit = {
+    db.withConnection(autocommit = true) { implicit conn =>
+      val request = SQL"""
+        INSERT INTO contribution_metadata(
+          contributionid,
+          created,
+          email,
+          ophanid,
+          abtests,
+          cmp,
+          intcmp
+        ) VALUES (
+          ${pmd.contributionId}::uuid,
+          ${pmd.created},
+          ${pmd.email},
+          ${pmd.ophanId},
+          ${pmd.abTests},
+          ${pmd.cmp},
+          ${pmd.intCmp}
+        ) ON CONFLICT(contributionId) DO
+        UPDATE SET
+          contributionid = excluded.contributionid,
+          created = excluded.created,
+          email = excluded.email,
+          ophanid = excluded.ophanid,
+          abtests = excluded.abtests,
+          cmp = excluded.cmp,
+          intcmp = excluded.intcmp"""
+      request.execute()
+    }
+  }
+
+  def insertContributor(contributor: Contributor): Unit = {
+    db.withConnection(autocommit = true) { implicit conn =>
```

Comment:
I'm not keen on us directly accessing an SQL database from contributions. I'd be happier if we had some kind of API endpoint or queue instead. Also, directly including the SQL logic here might make things harder when we want a secondary data store to handle end user requests. (ie history/contributions status)